### ISSUE TITLE
Use WebRTC stats for client-side reporting of network quality

### ISSUE THREAD

**Issue by ffdixon** [bigbluebutton/bigbluebutton#3376]
There is an emerging standard for the browser to make available stats on the quality of a WebRTC connection.  See

   https://w3c.github.io/webrtc-stats/

Investigate incorporating these stats into the BigBlueButton client.

**Comment by dipenpatel235** [comment]
Thanks ffdixon.
i will also work on that and if i am able to integrate that so i will give you that change.
otherwise u are already expert on that.

**Comment by ffdixon** [comment]
With the increasing support with webRTC stats
 
   http://iswebrtcreadyyet.com/

and the information provided

  https://www.callstats.io/2015/07/06/basics-webrtc-getstats-api/

we could give the user some client-side feedback on the "signal strength" of real-time transfer protocol (RTP) connecting using jitter, packetsLost, and packetsDiscarded.

We could then show that to the user using a series of bars when connected to WebRTC audio, such as

![image](https://cloud.githubusercontent.com/assets/290351/24406497/0b56cc6a-138e-11e7-98cc-fbcb37fc12a0.png)

We would not show these bars when connected to Flash audio -- only WebRTC audio and only on browsers that support the getStats API for WebRTC.

**Comment by dipenpatel235** [comment]
Thanks you so much Fred Dixon.
Nice to see this feature.

Thanks.
Dipen patel


On Mar 28, 2017 6:45 PM, "Fred Dixon" <notifications@github.com> wrote:

With the increasing support with webRTC stats

http://iswebrtcreadyyet.com/

and the information provided

https://www.callstats.io/2015/07/06/basics-webrtc-getstats-api/

we could give the user some client-side feedback on the "signal strength"
of real-time transfer protocol (RTP) connecting using jitter, packetsLost,
and packetsDiscarded.

We could then show that to the user using a series of bars when connected
to WebRTC audio, such as

[image: image]
<https://cloud.githubusercontent.com/assets/290351/24406497/0b56cc6a-138e-11e7-98cc-fbcb37fc12a0.png>

We would not show these bars when connected to Flash audio -- only WebRTC
audio and only on browsers that support the getStats API for WebRTC.

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub
<https://github.com/bigbluebutton/bigbluebutton/issues/3376#issuecomment-289765653>,
or mute the thread

**Comment by capilkey** [comment]
I don't know that we can add images to a button like that. That's also where text goes.

**Comment by dipenpatel235** [comment]
hello
i can't see this feature in 1.1 release .
have u added this feature in 1.1 release ?

Thanks

**Comment by ffdixon** [comment]
This feature is not in this release -- it's slated for inclusion in BigBlueButton 2.0 (we'll be making an announcement on the features planned for 2.0 to https://groups.google.com/forum/#!forum/bigbluebutton-dev shortly, as well as creating issues for each enhancement and tagging them 2.0)

**Comment by dipenpatel235** [comment]
Ok..thanks

**Comment by ffdixon** [comment]
There are more API calls we can make with Chrome to determine network quality for user, see

   https://googlechrome.github.io/samples/network-information/index.html

Tried this on a train where the internet was intermittent and got the following results

~~~
  // Network type that browser uses
  console.log('         type: ' + navigator.connection.type);
  // Effective bandwidth estimate
  console.log('     downlink: ' + navigator.connection.downlink + 'Mb/s');
  // Effective round-trip time estimate
  console.log('          rtt: ' + navigator.connection.rtt + 'ms');
  // Upper bound on the downlink speed of the first network hop
  console.log('  downlinkMax: ' + navigator.connection.downlinkMax + 'Mb/s');
  // Effective connection type determined using a combination of recently
  // observed rtt and downlink values: ' +
  console.log('effectiveType: ' + navigator.connection.effectiveType);
  // True if the user has requested a reduced data usage mode from the user
  // agent.
  console.log('     saveData: ' + navigator.connection.saveData);
~~~

I got the following results

~~~
          type: unknown
      downlink: 0.75Mb/s
           rtt: 550ms
   downlinkMax: InfinityMb/s
 effectiveType: 3g
      saveData: false
~~~

Anything that shows 3g is not enough.  These APIs look to be only available on Chrome

   http://wicg.github.io/netinfo/

but if the user is running Chrome we should use the to help determine if the user's network is sufficient for BigBlueButton.
